ui/servsettings.py

```diff
@@ -360,3 +386,322 @@ async def get_content(self):
+            ephemeral=True,
+        )
+        try:
+            input_msg: disnake.Message = await self.bot.wait_for(
+                "message",
+                timeout=60,
+                check=lambda msg: msg.author == interaction.author and msg.channel.id == interaction.channel_id,
+            )
+            if not 1 <= int(input_msg.content) <= 10:
+                raise ValueError
+            self.settings.randchar_num = int(input_msg.content)
+            if self.settings.randchar_num != len(self.settings.randchar_stat_names):
+                self.settings.randchar_straight = False
+            with suppress(disnake.HTTPException):
+                await input_msg.delete()
+        except (ValueError, asyncio.TimeoutError):
+            await interaction.send(
+                "No valid number of stats found. Press `Set Number of Stats` to try again.", ephemeral=True
+            )
+        else:
+            await self.commit_settings()
+            await interaction.send("Your number of stats have been updated.", ephemeral=True)
+        finally:
+            button.disabled = False
+            await self.refresh_content(interaction)
+
+    @disnake.ui.button(label="Toggle Assign Stats", style=disnake.ButtonStyle.primary)
+    async def toggle_straight(self, button: disnake.ui.Button, interaction: disnake.Interaction):
+        self.settings.randchar_straight = not self.settings.randchar_straight
+        if self.settings.randchar_straight:
+            button.disabled = True
+            await self.refresh_content(interaction)
+            await interaction.send(
+                "Choose the stat names to automatically assign the rolled stats to, separated by commas.\n"
+                "If you wish to use the default stats, respond with 'default'. This will only work if your number "
+                "of stats is 6.",
+                ephemeral=True,
+            )
+            try:
+                input_msg: disnake.Message = await self.bot.wait_for(
+                    "message",
+                    timeout=60,
+                    check=lambda msg: msg.author == interaction.author and msg.channel.id == interaction.channel_id,
+                )
+                message = input_msg.content
+                if message.lower() == "default":
+                    stats = [stat.upper() for stat in STAT_ABBREVIATIONS]
+                else:
+                    stats = message.replace(", ", ",").split(",")
+                if len(stats) != self.settings.randchar_num:
+                    print(stats, len(stats), self.settings.randchar_num)
+                    raise ValueError
+                self.settings.randchar_stat_names = stats
+                with suppress(disnake.HTTPException):
+                    await input_msg.delete()
+            except (ValueError, asyncio.TimeoutError):
+                self.settings.randchar_straight = not self.settings.randchar_straight
```

Comment:
```suggestion
                self.settings.randchar_straight = False
```
Explicit is better than implicit here

Reply:
Good call